compiler-plugin/src/main/scala/scalapb/compiler/ConstructorField.scala

```diff
@@ -24,4 +24,7 @@ object ConstructorField {
     default = Some(C.UnknownFieldSetEmpty),
```

Comment:
Let's make this `val` a `private[this]`. It's going be a bit hacky for now to detect whether a constructor field is the unknown fields. Instead of comparing by name in the generator, can you add a method `isUnknownField` in `ConstructorField` that compares the index with the number `Int.MaxValue`... This at least would ensure that the details are in a single place (and we avoid potential false positive matches)

Reply:
I removed it as it was not used in the code base anymore.
Instead added a value for the `UnknownFieldsIndex` that I am using in the method `isUnknownFields` as you suggested.